server/container/guice/memory-guice/src/main/java/org/apache/james/modules/server/MemoryMailQueueFactory.java

```diff
@@ -77,7 +80,11 @@ public void enQueue(Mail mail, long delay, TimeUnit unit) throws MailQueueExcept
 
         @Override
         public void enQueue(Mail mail) throws MailQueueException {
-            mailItems.addFirst(new MemoryMailQueueItem(mail));
+            try {
+                mailItems.addFirst(new MemoryMailQueueItem(new MailImpl(mail)));
```

Comment:
No.
Initial mail will be disposed in his scope. You have dispose instructions after enqueues (hence my troubles)
The copied mail will be disposed after his upcoming processing